### ISSUE TITLE
Make server debuggable using VSCode

### ISSUE THREAD

**Issue by mattkrick** [ParabolInc/parabol#3958]
## Issue
We are not able to debug the server in local or production environment. Developer productivity can be doubled if we are capable of that.

Quote from Matt:
> We must've had. regression somewhere & server debugging no longer works.

### Acceptance Criteria
- [] Developer can debug the server locally using breakpoints in VS Code
~~Developer can debug staging server in VS Code through SSH tunneling~~ (don't need now)
- [] Documentation for how to setup the process are in Notion

- **Estimated effort:** 12 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/parabol/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by tiffanyhan** [comment]
@mattkrick  i have pinpointed the local server debug problem. config in `scripts/webpack/dev.servers.config.js` needs to be set to `devtool: 'eval-source-map'` instead of `devtool: 'eval'`. after making the change, i am able to set breakpoints in `*.ts` files and use the debug console interactively.

shall i make this change in a PR? i have not looked into debugging prod remotely yet.

**Comment by tianrunhe** [comment]
Woo 🦅👀 there! Yeah I'll say go ahead and do a PR. Also if you could start a doc in Notion, I will chime in.

For the remote server debugging, on a second thought I don't think it's a good idea to do it against production server as we probably don't wanna users to halt on our debugging... For staging server it might make more sense. With local debugging available now I can try to make that happen with SSH tunnel.

**Comment by tiffanyhan** [comment]
Sounds good, @tianrunhe ! [PR made](https://github.com/ParabolInc/parabol/pull/3982) and [doc in Notion](https://www.notion.so/parabol/How-to-Debug-the-server-with-breakpoints-in-VSCode-5025cf93503d45868ebbf8d45c5d6ef8) created.

**Comment by tianrunhe** [comment]
I verified it works for me. PR approved! Doc also looks great. I really like screenshots! Thank you!

**Comment by tianrunhe** [comment]
I looked around a little bit on [remote debugging](https://nodejs.org/en/docs/guides/debugging-getting-started/#enabling-remote-debugging-scenarios) and found it might not worth the effort **now** to set it up, let alone potential security risk it might expose. We don't want to remote debug production environment for sure. For staging, we should be able to reproduce the issue, if any, on dev machine by [copying DB to local](https://www.notion.so/parabol/How-to-restore-a-remote-db-backup-to-local-43cdbfd072414f26b3e11e455425f300). Therefore I revised the acceptance criteria. This issue can now be resolved after PR is merged.
